DotNet/LinkAdmin.BFF/Presentation/Endpoints/AuthEndpoints.cs

```diff
@@ -87,8 +89,34 @@ public IResult Login(HttpContext context)
 
         public IResult GetUser(HttpContext context)
         {
-            var user = context.User;
-            return Results.Ok(user.Claims.Select(x => new { x.Type, x.Value }).ToList());
+            //get claims principal
+            if (context.User.Identity is not ClaimsIdentity claimsIdentity)
+            {
+                return Results.Problem("No claims found in the current user identity", statusCode: StatusCodes.Status500InternalServerError);
+            }
+
+            // Define the claim types to keep
+            var allowedClaims = new HashSet<string> {
```

Comment:
Consider abstracting `allowedClaims` into something that can be re-used between the two classes.